test/test_integration_resource.py

```diff
@@ -0,0 +1,182 @@
+                    new_api_key = line.split(" ")[1]
+
+        assert new_api_key.strip() == extract_api_key_from_message, "the API keys are not the same!"
+
+    @integration_test
+    def test_user_rotate_api_key_with_user_provided_rotates_user_api_key(self):
+        some_user_api_key = self.invoke_cli(self.cli_auth_params,
+               ['user', 'rotate-api-key', '-i', 'someuser'])
+        extract_api_key_from_message = some_user_api_key.split(":")[1].strip()
+
+        self.assertIn("API key for 'someuser' was successfully rotated", some_user_api_key)
+
+        # verify user can login with their new API key
+        self.invoke_cli(self.cli_auth_params,
+                        ['login', '-n', 'someuser', '-p', extract_api_key_from_message], exit_code=0)
+
+    '''
+    Validates that an unprivileged user cannot rotate another user's API key.
+    1. Rotate user's API key (using the admin's access token) to be able to login as them
+    2. Login as that user and attempt to rotate another's API key
+    '''
+    @integration_test
+    def test_unprivileged_user_cannot_rotate_anothers_api_key(self):
+        unprivileged_api_key = self.invoke_cli(self.cli_auth_params,
+                                               ['user', 'rotate-api-key', '-i', 'someunprivilegeduser'], exit_code=0)
+        # extract the API key we get from the CLI success message
+        extract_api_key_from_message = unprivileged_api_key.split(":")[1].strip()
+        print(extract_api_key_from_message)
+        # verify user can login with their new API key
+        self.invoke_cli(self.cli_auth_params,
+                        ['login', '-n', 'someunprivilegeduser', '-p', extract_api_key_from_message])
+
+        # Verify that an unprivileged user cannot rotate an another's API key
+        attempt_to_rotate_user_key = self.invoke_cli(self.cli_auth_params,
+               ['user', 'rotate-api-key', '-i', 'someuser'], exit_code=1)
+        self.assertIn("401 Client Error", attempt_to_rotate_user_key)
+
+        # Verify that a user cannot rotate an admin's API key
+        attempt_to_rotate_admin_key = self.invoke_cli(self.cli_auth_params,
+               ['user', 'rotate-api-key', '-i', 'admin'], exit_code=1)
+        self.assertIn("500 Server Error", attempt_to_rotate_admin_key)
+
+    def test_user_rotate_api_key_without_flag_returns_error(self):
+        with redirect_stderr(self.capture_stream):
+            self.invoke_cli(self.cli_auth_params,
+                            ['user', 'rotate-api-key', 'someinput'], exit_code=1)
+        self.assertIn('Error', self.capture_stream.getvalue())
+
+    def test_user_rotate_api_key_flag_without_value_returns_error(self):
+        with redirect_stderr(self.capture_stream):
+            self.invoke_cli(self.cli_auth_params,
+                            ['user', 'rotate-api-key', '-i'], exit_code=1)
+
+        self.assertIn('Error', self.capture_stream.getvalue())
```

Comment:
maybe we need to be more specific here ?

Reply:
you are correct. updated